src/granite/querying.cr

```diff
@@ -28,13 +28,14 @@ module Granite::Querying
   # that you are using so you are not restricted or dummied down to support a
   # DSL.
   # Lazy load prevent running unnecessary queries from unused variables.
-  def all(clause = "", params = [] of Granite::Columns::Type)
+  def all(clause = "", params = [] of Granite::Columns::Type, use_primary_adapter = true)
+    switch_to_writer_adapter if use_primary_adapter == true
     Collection(self).new(->{ raw_all(clause, params) })
   end
 
   # First adds a `LIMIT 1` clause to the query and returns the first result
   def first(clause = "", params = [] of Granite::Columns::Type)
-    all([clause.strip, "LIMIT 1"].join(" "), params).first?
+    all([clause.strip, "LIMIT 1"].join(" "), params, false).first?
```

Comment:
This should be `true` when performing this read action on the same model that recently had a `write` action performed. I don't think I've seen anything for that logic yet

Reply:
The switch only happens if > 2 seconds have passed since last write operation.

Reply:
The logic for handling switching is in Granite::Base.

We use the primary database connection if the user calls this function directly. Any other functions that call this function actually use the reader adapter. The system schedules a switch to the reader adapter after any write operations. So in those cases, we don't need to explicitly set the adapter to the writer adapter because the system will only change the adapter 2 seconds after a write operation. We use whatever operator is available. Before 2 seconds we use writer adapter. After 2 seconds it automatically uses reader adapter. The use_primary = true is just to make sure we use the writer adapter when the user calls this method directly because that is what the user would expect.